src/permission/net_permission.cc

```diff
@@ -0,0 +1,172 @@
+                          const std::vector<std::string>& allow,
+                          PermissionScope scope) {
+  // For Debug
+  auto cleanup = OnScopeLeave([&]() { Print(); });
+  using std::string_view_literals::operator""sv;
+  for (const std::string& res : allow) {
+    // Allow multiple item in one flag, like --allow-net-udp=127.0.0.1,localhost
+    const std::vector<std::string_view> addresses = SplitString(res, ","sv);
+    for (const auto& address : addresses) {
+      switch (scope) {
+        case PermissionScope::kNetUDP:
+          // address is like *, *:*, host, host:*, host:port, *:port,
+          // host/netmask:port
+          if (address == "*"sv || address == "*:*"sv) {
+            deny_all_udp_ = false;
+            allow_all_udp_ = true;
+            granted_udp_.clear();
+            return;
+          }
+          GrantAccess(scope, address);
+          continue;
+        default:
+          continue;
+      }
+    }
+  }
+}
+
+void NetPermission::GrantAccess(PermissionScope scope,
+                                const std::string_view& param) {
+  std::unique_ptr<PermissionAddressInfo> result = ParseAddress(param);
+  std::bitset<IPV6_BIT_LEN> ip;
+  std::bitset<IPV6_BIT_LEN> netmask;
+  std::unique_ptr<NetMaskNode> net_mask_node;
+  int ip_version = IsIP(result->address);
+
+  if (ip_version > 0 && !result->netmask.empty()) {
+    if (IsDigit(result->netmask)) {  // 127.0.0.1/24
+      int netmask_len = std::stoi(result->netmask);
+      CHECK_GE(netmask_len, 0);
+      int position = ip_version == 6 ? IPV6_BIT_LEN : IPV4_BIT_LEN;
+      while (netmask_len--) {
+        netmask.set(--position, true);
+      }
+    } else {  // 127.0.0.1/255.255.255.0
+      CHECK(ConvertIPToBitset(&netmask, result->netmask, ip_version));
+    }
+    CHECK(ConvertIPToBitset(&ip, result->address, ip_version));
+    std::bitset<IPV6_BIT_LEN> network = netmask & ip;
+    net_mask_node = std::make_unique<NetMaskNode>(netmask, network);
+  }
+  auto node = std::make_unique<NetNode>(
+      result->address, result->port, std::move(net_mask_node));
+  granted_udp_.push_back(std::move(node));
+  deny_all_udp_ = false;
+}
+
+bool NetPermission::is_granted(Environment* env,
+                               PermissionScope perm,
+                               const std::string_view& param = "") const {
+  switch (perm) {
+    case PermissionScope::kNetUDP:
+      return !deny_all_udp_ &&
+             (allow_all_udp_ || CheckoutResource(granted_udp_, param));
+    default:
+      return false;
+  }
+}
+
+bool NetPermission::CheckoutResource(
+    const std::vector<std::unique_ptr<NetNode>>& list,
+    const std::string_view& param) const {
+  if (param.empty()) {
+    return false;
+  }
+  using std::string_view_literals::operator""sv;
+  const std::vector<std::string_view> address_port = SplitString(param, "/"sv);
+  int size = address_port.size();
+  CHECK(size == 1 || size == 2);
+  std::string address;
+  std::string port;
```

Comment:
This makes a copy. You can keep it as `std::string_view` until you explicitly need to copy.